packages/compass-aggregations/src/utils/stage.js

```diff
@@ -86,20 +95,20 @@ export function getStageOperator(stage) {
  * @see {@link https://www.mongodb.com/docs/atlas/data-federation/supported-unsupported/pipeline/out/#syntax}
  *
  * @param {string} namespace
- * @param {unknown} stage
+ * @param {import('mongodb').Document} stage
  * @returns {string}
  */
  export function getDestinationNamespaceFromStage(namespace, stage) {
   const stageOperator = getStageOperator(stage);
   const stageValue = stage[stageOperator];
   const { database } = toNS(namespace);
   if (stageOperator === '$merge') {
-    const ns = typeof stage === 'string' ? stageValue : stageValue.into;
-    if (ns.atlas) {
+    if (stageValue.atlas) {
       // TODO: Not handled currently and we need some time to figure out how to
       // handle it so just skipping for now
       return null;
     }
+    const ns = typeof stageValue === 'string' ? stageValue : stageValue.into;
```

Comment:
I am not sure why this was moved down. From the documentation, `atlas` is the key on `into` property, not on the stage description itself, [see the documentation linked in the method description](https://www.mongodb.com/docs/atlas/data-federation/supported-unsupported/pipeline/merge/#syntax)

Reply:
ah damn, i was following [docs on aggregation page](https://www.mongodb.com/docs/manual/reference/operator/aggregation/merge/).

Reply:
fixed.